app/assets/javascripts/courses.js

```diff
@@ -2,367 +2,370 @@ var ready;
+          if (attrName != otherSortOptions[i]) {
+
+            // Compare by that stat
+            retVal = compareProfessors(professorA, professorB, otherSortOptions[i]);
+
+            // stop if found a tie breaker
+            if (retVal != 0) {
+              break;
+            }
+            // if three-way tie, return 0
+            if (i == 2 && retVal == 0) {
+              return 0;
+            }
+          }
+        }
+      }
+
+      return retVal;
+
+    }));
+  }
+
+  // Compares two professor panel elements by a given stat (id)
+  function compareProfessors(professorA, professorB, stat) {
+    // Get stat to sort by (rating, difficulty, gpa)
+    var aVal = parseFloat($(professorA).find(stat).text()),
+      bVal = parseFloat($(professorB).find(stat).text()),
+      retVal = bVal - aVal;
+
+    // Handle no stat
+    if (isNaN(aVal) && !isNaN(bVal)) {
+      return 1;
+    } else if (isNaN(bVal) && !isNaN(aVal)) {
+      return -1;
+    } else if (isNaN(bVal) && isNaN(aVal)) {
+      retVal = 0;
+    }
+
+    // Sort difficulty ascending
+    if (stat == ".course-difficulty") {
+      retVal = retVal * -1;
+    }
+
+    return retVal;
+  }
+
+
+  // Sort professors by stats
+  $('#prof-sort').change(function() {
+
+    // id of stat to sort by    
+    var sortString = ".course-" + $(this).find('.active')[0].id,
+      // class of which panels are displayed (current semester or all)
+      selectorString = $("#all").parent().hasClass("active") ? ".prof-panel.all" : ".prof-panel.current",
+      // how many panels there are (to know when to trigger the next animation)
+      numPanels = $(selectorString).length,
+      slidPanels = 0,
+      // sorted list of professors
+      profList = sortProfessors($(".prof-panel"), sortString);
+    // if the number of panels displayed is small enough to have smooth animations, animate the change
+    if (numPanels < 50) {
+      // slide up all the panels. then, on complete, add the sorted ones and slide down what is needed.
+      $(selectorString).slideUp(350, function() {
+        slidPanels++;
+        if (slidPanels == numPanels) {
+          $(".prof-panel-container").empty();
+          $(".prof-panel-container").append(profList);
+          $(selectorString).slideDown(350);
+        }
+      });
+    } else {
+      // else just pop on the sorted list
+      $(".prof-panel-container").empty();
+      $(".prof-panel-container").append(profList);
+    }
+  });
+
+
+  // If this is not the section page (no grade wheel),
+  // then the rest of this code is not needed
+  if (!$('.course-grades').length) {
+    return
+  }
+
+  window.onscroll = function(ev) {
+    if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight && $('.reviews-box').children('.hidden').length > 0) {
+      $('.wheel-loader').removeClass('hidden');
+      setTimeout(function () {
+        appendReviews();
+        $('.wheel-loader').addClass('hidden');
+      }, 800);
```

Comment:
Not sure if this delay is better or worse for usability. To me it felt more laggy and jumpy
@aw3as @emumoo @DC00 @Apoorva-A Thoughts?